src/WebdriverClassicDriver.php

```diff
@@ -11,11 +11,1194 @@
+    {
+        try {
+            $select = new WebDriverSelect($element);
+            if (!$multiple && $select->isMultiple()) {
+                $select->deselectAll();
+            }
+            try {
+                $select->selectByValue($value);
+            } catch (NoSuchElementException $e) {
+                $select->selectByVisibleText($value);
+            }
+        } catch (Throwable $e) {
+            $message = sprintf(
+                'Cannot select option "%s" of "%s": %s',
+                $value,
+                $element->getAttribute('name'),
+                $e->getMessage(),
+            );
+            throw new DriverException($message, 0, $e);
+        }
+    }
+
+    /**
+     * Deselects all options of a multiple select
+     *
+     * Note: this implementation does not trigger a change event after deselecting the elements.
+     *
+     * @throws DriverException
+     */
+    private function deselectAllOptions(RemoteWebElement $element): void
+    {
+        try {
+            (new WebDriverSelect($element))->deselectAll();
+        } catch (Throwable $e) {
+            $message = sprintf(
+                'Cannot deselect all options of "%s": %s',
+                $element->getAttribute('name'),
+                $e->getMessage()
+            );
+            throw new DriverException($message, 0, $e);
+        }
+    }
+
+    /**
+     * @throws DriverException
+     */
+    private function ensureInputType(
+        RemoteWebElement $element,
+        #[Language('XPath')]
+        string $xpath,
+        string $type,
+        string $action
+    ): void {
+        if ($element->getTagName() !== 'input' || $element->getAttribute('type') !== $type) {
+            throw new DriverException(
+                "Impossible to $action the element with XPath \"$xpath\" as it is not a $type input"
+            );
+        }
+    }
+
+    /**
+     * @throws DriverException
+     */
+    private function trigger(
+        #[Language('XPath')]
+        string $xpath,
+        string $event,
+        #[Language('JSON')]
+        string $options = '{}'
+    ): void {
+        $this->withSyn()->executeJsOnXpath($xpath, "window.syn.trigger(arguments[0], '$event', $options)");
+    }
+
+    /**
+     * @param mixed $value
+     * @throws DriverException
+     */
+    private function jsonEncode($value, string $action, string $field): string
+    {
+        try {
+            return json_encode($value, JSON_THROW_ON_ERROR);
+        } catch (JsonException $e) {
+            throw new DriverException("Cannot $action, $field not serializable: {$e->getMessage()}", 0, $e);
+        }
+    }
+
+    /**
+     * @param mixed $value
+     * @throws DriverException
+     * @phpstan-assert string $value
+     * @todo When switching to PHP 8, this can be replaced with `is_string(..) or throw new DriverException(..);`
+     */
+    private function assertString($value, string $message): void
```

Comment:
As this method is used only in contexts that do not require an expression (which would indeed not be possible in PHP 7.4), I suggest inlining it.
There is no reason to add this method with a TODO asking to replace it with an expression when we can already replace it.